### PR TITLE
Issue #20135: VAT Taxation enhancements

### PR DESCRIPTION
**Pull request by andersdd**
Initial commit of VAT taxation enhancements.  These changes stand on their own but further changes required for PO Freight Taxation.   I've added a Tax Return report - but not entirely convinced they are sufficiently different from Tax History to warrant the additional screen / report.

requires xtuple/xtuple#2495

### CODE REVIEW

File: guiclient/dspTaxReturn.cpp [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  if (valid)
+    _purchases->setChecked(param.toBool());
+  
+  param = pParams.value("startDate", &valid);
+  if (valid)
+    _dates->setStartDate(param.toDate());
+    
+  param = pParams.value("endDate", &valid);
+  if (valid)
+    _dates->setStartDate(param.toDate());
```
Comment (gilmoskowitz) [inline guiclient/dspTaxReturn.cpp]: `_dates->setEndDate()`?

File: guiclient/miscCheck.cpp [hunk 2]
```diff
@@ -394,3 +439,30 @@ void miscCheck::sCreditMemoSelected()
...
+  calcTax.exec();
+  if (ErrorReporter::error(QtCriticalMsg, this,
+                                tr("Tax Calculation"),
+                                calcTax, __FILE__, __LINE__))
+    return; 
+  else
+  {
+    if (calcTax.first())
+      _tax->setLocalValue(calcTax.value("tax").toDouble());
+  }
```
Comment (gilmoskowitz) [inline guiclient/miscCheck.cpp]: What's the rationale for inverting the normal flow of
```
if (q.first()) {
  doStuff;
} else if (ErrorReporter::error(...)) {
  return;
}
```
?

**Comment by andersdd** [comment]
@gilmoskowitz had an epiphany yesterday about the Misc Voucher and our discussions about tax being applicable or not depending on the tax type.  I have made some assumptions in this current code, but think I can improve the logic and flexibility with some minor changes.  These changes also work with the freight tax subsequent work - which is where the idea came from.

**Comment by gilmoskowitz** [comment]
@andersdd - OK. Let me know when I should review this again.

**Pull request closed without merging**